packages/block-library/src/file/index.php

```diff
@@ -64,8 +64,8 @@ function register_block_core_file() {
 	);
 
 	wp_register_script_module(
-		'@wordpress/block-library/file-block',
-		gutenberg_url( '/build/interactivity/file.min.js' ),
+		'@wordpress/block-library/file',
+		defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN ? gutenberg_url( '/build/interactivity/file.min.js' ) : includes_url( 'blocks/file/view.min.js' ),
```

Comment:
Just checking my understanding here. What's the long term vision for these `wp_register_script_module` calls? Are they going to be done automatically using `block.json` or if we need conditionals, maybe instead we should have generic calls like `wp_hydrate_block( 'core/file' )` or something where the actual files are not hard-coded by guessed from block.json or something.

Reply:
As far as I remember @sirreal is working on a viewModule and viewScriptModule feature in block json. So should be done automatically.

Reply:
Yes. In the short term, we should replace them with `viewScriptModule` entries in the `block.json`. Then, at some point, I'd like to resume our work on behaviors where a single block can have a set of behaviors defined, and depending on the active ones, WordPress will enqueue the required JS and CSS files for those behaviors.

Reply:
or viewScript?

Reply:
In the beginning, we called it the Modules API but then everything was renamed to "script modules":

- https://github.com/WordPress/wordpress-develop/pull/5818#issuecomment-1887207690
- https://github.com/WordPress/wordpress-develop/pull/5869